### PR TITLE
Goyaml - move to a fork instead of patch file

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

The following commit removes the patch file for go-yaml race
issues and instead depends on a fork with the fixes applied. This
way we can make sure that the source of juju is correct, because
the source code of go-yaml is applied at source.

## QA steps

 1. Ensure that all tests pass
 1. Run race detector to see that races have disappeared for yaml.

## Documentation changes

Developers of juju should no longer provide patch files for upstream
changes and instead use `dep` source re-writing to do the same job.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1795007

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/YdrRLDqS/789-remove-goyaml-patch-files-and-favour-dep-re-writing

**Comment by mitechie** [comment]
Looks like it failed due to the change: 

    05:58:23 network/netplan/netplan.go:344:19:warning: UnmarshalStrict not declared by package yaml (unconvert)
    05:58:23 core/raftlease/fsm.go:378:18:warning: NewEncoder not declared by package yaml (unconvert)
    05:58:23 core/raftlease/fsm.go:320:18:warning: NewDecoder not declared by package yaml (unconvert)
    05:58:23 cmd/juju/application/bundlediff.go:146:18:warning: NewEncoder not declared by package yaml (unconvert)

**Comment by manadart** [comment]
Depends on https://github.com/juju/yaml/pull/1, and an updated hash for that lib.

File: patches/goyaml-pr-241.diff [hunk 1]
```diff
@@ -1,31 +0,0 @@
-This is the diff from PR #241 on goyaml.
-When this file is removed, also remove the Skip from the test
-featuretests/cmdjuju_test.go TestServiceGetWeirdYAML.
```
Comment (manadart) [inline patches/goyaml-pr-241.diff]: We should remember to do this when we have _yaml_ at the required hash.

Comment (SimonRichardson) [reply patches/goyaml-pr-241.diff]: @manadart good shout, totally missed this.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**